Replace `block:` with `rules:` configuration for changelog filtering (#2693)
elastic/docs-builder: docs/contribute/changelog.md

Each field supports **exclude** (block if matches) or **include** (block if doesn't match) semantics. You cannot mix both for the same field.

For multi-valued fields (labels, areas), you can control the matching mode:
- `any` (default): match if ANY item matches the list
- `all`: match only if ALL items match the list

:::{note}
You can define rules at the global level (applies to all products) or for specific products.
Product-specific rules **override** the global rules entirely—they do not merge.
If you define a product-specific `publish` rule, you must re-state any global rules that you also want applied for that product.
:::

Refer to [changelog.example.yml](https://github.com/elastic/docs-builder/blob/main/config/changelog.example.yml) and an [example usage](#example-block-label).

### Rules reference

The `rules:` section supports the following options:

#### `rules.match`

Global match default for multi-valued fields (labels, areas). Inherited by `create`, `publish`, and all product overrides.

| Value | Description |
|-------|-------------|

> | `any` (default) | Match if ANY item in the entry matches an item in the configured list |

Reviewer: IMO "changelog field" is clearer than "entry" for those of us who aren't as familiar with YAML structure:
```suggestion
| `any` (default) | Match if ANY item in the changelog field matches an item in the configured list |
```

> | `all` | Match only if ALL items in the entry match items in the configured list |

Reviewer: Ditto
```suggestion
| `all` | Match only if ALL items in the changelog field match items in the configured list |
```

#### `rules.create`

> Controls which PRs generate changelog entries. Evaluated when running `docs-builder changelog add` with `--prs`.

Reviewer: ```suggestion
Controls which PRs generate changelogs. Evaluated when running `docs-builder changelog add` with `--prs`.
```

| Option | Type | Description |
|--------|------|-------------|
| `exclude` | string | Comma-separated labels that prevent changelog creation. A PR with any matching label is skipped. |
| `include` | string | Comma-separated labels required for changelog creation. A PR without any matching label is skipped. |
| `match` | string | Override `rules.match` for create rules. Values: `any`, `all`. |
| `products` | map | Product-specific create rules (see below). |

You cannot specify both `exclude` and `include`.

**Product-specific create rules** (`rules.create.products`):

Product keys can be a single product ID or a comma-separated list of product IDs (for example, `'elasticsearch, kibana'`).
Each product override supports the same `exclude`, `include`, and `match` options.
Product-specific rules **override** the global create rules entirely.

```yaml
rules:
  create:
    exclude: ">non-issue"
    products:
      'elasticsearch, kibana':
        exclude: ">test"
      cloud-serverless:
        exclude: ">non-issue, ILM"
```

#### `rules.publish`

> Controls which entries appear in rendered output. Evaluated when running `docs-builder changelog render` or using the `{changelog}` directive.

Reviewer: IMO "entries" is unclear since it could be interpreted as a part of a changelog rather than the whole changelog (e.g. per terminology in https://yaml.org/spec/1.2.2/#chapter-2-language-overview)
```suggestion
Controls which changelogs appear in rendered output. Evaluated when running `docs-builder changelog render` or using the `{changelog}` directive.
```

| Option | Type | Description |
|--------|------|-------------|
| `exclude_types` | list | Entry types to hide from output |
| `include_types` | list | Only these entry types are shown; all others are hidden |
| `exclude_areas` | list | Entry areas to hide from output |
| `include_areas` | list | Only entries with these areas are shown; all others are hidden |
| `match_areas` | string | Override `rules.match` for area matching. Values: `any`, `all`. |

> | `products` | map | Product-specific publish rules (see below). |

Reviewer: Another accessibility thingy
```suggestion
| `products` | map | Product-specific publish rules (refer to following section). |
```

#### Release highlights

The `highlight` field allows you to mark changelogs that should appear in a dedicated highlights page.
Highlights are most commonly used for major or minor version releases to draw attention to the most important changes.
Highlights can be any type of changelog entry (such as features, enhancements, or bug fixes).

> Example changelog entry with highlight:

Reviewer: ```suggestion
Example changelog with highlight:
```
